core/datatrak.c

```diff
@@ -561,11 +578,105 @@ static unsigned char *dt_dive_parser(unsigned char *runner, struct dive *dt_dive
 	free(devdata);
 	return NULL;
 }
+/*
+ * Parses the header of the .add file, returns the number of dives in
+ * the archive (must be the same than number of dives in .log file).
+ */
+static unsigned int wlog_header_parser (struct memblock *mem)
+{
+	int tmp;
+	unsigned char *runner = (unsigned char *) mem->buffer;
+	if (!runner)
+		return -1;
+	tmp = (runner[1] << 8) + runner[0];
+	if (tmp == 0x0252) {
+		runner += 8;
+		tmp = (runner[1] << 8) + runner[0];
+		return tmp;
+	} else {
+		fprintf(stderr, "Error, not a Wlog .add file\n");
+		return -1;
+	}
+}
+
+static void wlog_compl_parser(struct memblock *wl_mem, struct dive *dt_dive, int dcount)
+{
+	int strlong = 256, tmp = 0, offset = 12 + (dcount * 850),
+	    pos_weight =  offset + 256,
+	    pos_viz = offset + 258,
+	    pos_tank_init = offset + 266,
+	    pos_suit = offset + 268;
+	char *wlog_notes = NULL, *wlog_suit = NULL, *buffer = NULL;
+	unsigned char *runner = (unsigned char *) wl_mem->buffer;
+
+	/*
+	 * Extended notes string. Fixed length 256 bytes. 0 padded if not complete
+	 */
+	if (*(runner + offset)) {
+		wlog_notes = calloc(strlong + 1, 1);
+		wlog_notes = memcpy(wlog_notes, runner + offset, 256);
+		wlog_notes = to_utf8((unsigned char *) wlog_notes);
+	}
+	if (dt_dive->notes && wlog_notes) {
+		buffer = calloc (strlen(dt_dive->notes) + strlen(wlog_notes) + 1, 1);
+		sprintf(buffer, "%s%s", dt_dive->notes, wlog_notes);
+		free(dt_dive->notes);
+		dt_dive->notes = copy_string(buffer);
+	} else if (wlog_notes) {
+		dt_dive->notes = copy_string(wlog_notes);
+	}
+	free(buffer);
+	free(wlog_notes);
+
+	/*
+	 * Weight in Kg * 100
+	 */
+	tmp = (runner[pos_weight + 1] << 8) + runner[pos_weight];
```

Comment:
Perhaps create a helper function to access little endian 16-bit values? If that doesn't exist already.

Reply:
Yes I had one previously defined and widely used along the hole file. Just forgot about it.